scripts/utils/publishNewVersion.js

```diff
@@ -44,19 +46,52 @@ function projectSetup() {
 }
 
 function publishToNpm(npmTag) {
-  logSection('Lerna publish');
+  logSection('Yarn publish');
   const versionType = getReleaseVersionType();
   const dryRun = isDryRun();
   const skipNpm = isSkipNpm();
+
   if (dryRun) {
-    log('DRY RUN: Lerna-publishing without publishing to NPM');
+    log('DRY RUN: Publishing without pushing to NPM');
+  } else if (skipNpm) {
+    log('SKIP NPM is set: Skipping NPM publish');
   }
-  else if (skipNpm) {
-    log('SKIP NPM is set: Lerna-publishing without publishing to NPM');
+
+  // Bump versions in all packages
+  const preid = versionType.includes("pre") ? `--preid ${npmTag}` : '';
+
+  // Update version in root package.json
+  exec.execSync(`yarn version ${versionType} ${preid}`);
+
+  // Update versions in workspace packages
+  const packages = ['detox', 'detox-cli'];
```

Comment:
This is a bit too implicit to my taste, and is bound to go out of sync.
Is there a way this can be done differently in order to maintain a single source of truth (lerna.json style, or a custom property inside the various package.json files)?